Functions/Import/xASL_imp_DCM2NII_ConvertScan.m

```diff
@@ -162,35 +168,42 @@
 
 
     %% 4.7 Store JSON files
+    jsonFiles = {};
     if ~isempty(nii_files)
-        jsonFiles = nii_files;
+
         for iFile = 1:length(nii_files)
-            [Fpath, Ffile, ~] = fileparts(nii_files{iFile});
+            [Fpath, Ffile, ~] = xASL_fileparts(nii_files{iFile});
             newFile = xASL_adm_GetFileList(Fpath,['^' Ffile '.json$'],'FPList');
-            jsonFiles{iFile} = newFile{1};
+            if ~isempty(newFile)
+                jsonFiles{iFile} = newFile{1};
+            else
+                warning(['JSON sidecar missing for ' nii_files{iFile}]);
+            end
+        end
+        if ~isempty(jsonFiles)
+            [parms, x.modules.import.pathDcmDict] = xASL_imp_DCM2NII_Subject_StoreJSON(...
+                x.modules.import.imPar, jsonFiles, first_match, x.modules.import.settings.bUseDCMTK, x.modules.import.pathDcmDict);
         end
-        [parms, x.modules.import.pathDcmDict] = xASL_imp_DCM2NII_Subject_StoreJSON(...
-            x.modules.import.imPar, jsonFiles, first_match, x.modules.import.settings.bUseDCMTK, x.modules.import.pathDcmDict);
     end
 
-    %% 4.8 Sort ASL Volumes
     
-    % In case of GE or PARREC/a single NII ASL file loaded from PAR/REC, we need to 
-    % shuffle the dynamics from CCCC...LLLL order to CLCLCLCL... order.
-    [x,nii_files, summary_line, thisSubject.globalCounts] = ...
-        xASL_imp_DCM2NII_Subject_SortASLVolumes(x, thisSubject.globalCounts, scanpath, scan_name, nii_files, scanFields.iSubject, scanFields.iVisit, scanFields.iSession, scanFields.iScan);
-
-    % Correct nifti rescale slope if parms.RescaleSlopeOriginal =~1 but nii.dat.scl_slope==1 (this can happen in case
-    % of hidden scale slopes in private Philips header, that is dealt with by xASL_bids_Dicom2JSON but not by dcm2niiX.
-    if ~isempty(nii_files) && exist('parms','var')
-        [TempLine, PrintDICOMFields] = xASL_imp_AppendParmsParameters(parms);
-        summary_line = [summary_line TempLine];
-    else
-        PrintDICOMFields = [];
```

Comment:
It's wrong to remove PrintDICOMFields from here. And also since this is now inside another IF, you'll need to initialize PrintDICOMFields in the other branch as well.

Because it's an output variable and you will have it missing in certain cases -> error.

Reply:
I thought this was never used, this is why I removed it.

Reply:
We use it - it's the function output.

Reply:
You're right